# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 808c433c7774
# Full Hash: 808c433c77742953dea1f6e0456cd96a505e3615
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 09:46:07
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 7-4: Make `HTMLEditor::SetBlockElementAlign` stop touching `Selection` directly r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152977
# ==============================================================================

diff -r 511a7e64402c -r 808c433c7774 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Thu Aug 04 04:33:56 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Thu Aug 04 04:40:14 2022 +0000
@@ -2499,8 +2499,9 @@
    *                            descendants of aBlockOrHRElement.
    *                            If `NodeAndDescendantsExceptTable`, modifies
    *                            aBlockOrHRElement and its descendants.
-   */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
+   * @return                    A candidate point to put caret.
+   */
+  [[nodiscard]] MOZ_CAN_RUN_SCRIPT Result<EditorDOMPoint, nsresult>
   SetBlockElementAlign(Element& aBlockOrHRElement, const nsAString& aAlignType,
                        EditTarget aEditTarget);
 
